diff --git a/.gdb_history b/.gdb_history
new file mode 100644
index 0000000..bca70f3
--- /dev/null
+++ b/.gdb_history
@@ -0,0 +1 @@
+q
diff --git a/.gitignore b/.gitignore
old mode 100644
new mode 100755
diff --git a/LICENSE b/LICENSE
old mode 100644
new mode 100755
diff --git a/README.TXT b/README.TXT
old mode 100644
new mode 100755
diff --git a/README.md b/README.md
old mode 100644
new mode 100755
diff --git a/doomgeneric.sln b/doomgeneric.sln
old mode 100644
new mode 100755
diff --git a/doomgeneric/Makefile b/doomgeneric/Makefile
old mode 100644
new mode 100755
index 503e0dc..314bc2d
--- a/doomgeneric/Makefile
+++ b/doomgeneric/Makefile
@@ -12,17 +12,17 @@ else
 endif
 
 
-CC=clang  # gcc or g++
-CFLAGS+=-ggdb3 -Os
-LDFLAGS+=-Wl,--gc-sections
-CFLAGS+=-ggdb3 -Wall -DNORMALUNIX -DLINUX -DSNDSERV -D_DEFAULT_SOURCE # -DUSEASM
-LIBS+=-lm -lc -lX11
+CC=x86_64-ethereal-gcc  # gcc or g++
+CFLAGS+=-Os -g
+LDFLAGS+=-static
+CFLAGS+=-Wall -I/home/samuel/Desktop/reduceOS/build-output/sysroot/usr/include/ -I/home/samuel/Desktop/reduceOS/build-output/sysroot/usr/include/freetype2/ # -DUSEASM
+LIBS+=-lcelestial -lgraphics -lkeyboard -lfreetype -lstructures -lpolyhedron
 
 # subdirectory for objects
 OBJDIR=build
 OUTPUT=doomgeneric
 
-SRC_DOOM = dummy.o am_map.o doomdef.o doomstat.o dstrings.o d_event.o d_items.o d_iwad.o d_loop.o d_main.o d_mode.o d_net.o f_finale.o f_wipe.o g_game.o hu_lib.o hu_stuff.o info.o i_cdmus.o i_endoom.o i_joystick.o i_scale.o i_sound.o i_system.o i_timer.o memio.o m_argv.o m_bbox.o m_cheat.o m_config.o m_controls.o m_fixed.o m_menu.o m_misc.o m_random.o p_ceilng.o p_doors.o p_enemy.o p_floor.o p_inter.o p_lights.o p_map.o p_maputl.o p_mobj.o p_plats.o p_pspr.o p_saveg.o p_setup.o p_sight.o p_spec.o p_switch.o p_telept.o p_tick.o p_user.o r_bsp.o r_data.o r_draw.o r_main.o r_plane.o r_segs.o r_sky.o r_things.o sha1.o sounds.o statdump.o st_lib.o st_stuff.o s_sound.o tables.o v_video.o wi_stuff.o w_checksum.o w_file.o w_main.o w_wad.o z_zone.o w_file_stdc.o i_input.o i_video.o doomgeneric.o doomgeneric_xlib.o
+SRC_DOOM = dummy.o am_map.o doomdef.o doomstat.o dstrings.o d_event.o d_items.o d_iwad.o d_loop.o d_main.o d_mode.o d_net.o f_finale.o f_wipe.o g_game.o hu_lib.o hu_stuff.o info.o i_cdmus.o i_endoom.o i_joystick.o i_scale.o i_sound.o i_system.o i_timer.o memio.o m_argv.o m_bbox.o m_cheat.o m_config.o m_controls.o m_fixed.o m_menu.o m_misc.o m_random.o p_ceilng.o p_doors.o p_enemy.o p_floor.o p_inter.o p_lights.o p_map.o p_maputl.o p_mobj.o p_plats.o p_pspr.o p_saveg.o p_setup.o p_sight.o p_spec.o p_switch.o p_telept.o p_tick.o p_user.o r_bsp.o r_data.o r_draw.o r_main.o r_plane.o r_segs.o r_sky.o r_things.o sha1.o sounds.o statdump.o st_lib.o st_stuff.o s_sound.o tables.o v_video.o wi_stuff.o w_checksum.o w_file.o w_main.o w_wad.o z_zone.o w_file_stdc.o i_input.o i_video.o doomgeneric.o doomgeneric_ethereal.o
 OBJS += $(addprefix $(OBJDIR)/, $(SRC_DOOM))
 
 all:	 $(OUTPUT)
diff --git a/doomgeneric/Makefile.djgpp b/doomgeneric/Makefile.djgpp
old mode 100644
new mode 100755
diff --git a/doomgeneric/Makefile.emscripten b/doomgeneric/Makefile.emscripten
old mode 100644
new mode 100755
diff --git a/doomgeneric/Makefile.freebsd b/doomgeneric/Makefile.freebsd
old mode 100644
new mode 100755
diff --git a/doomgeneric/Makefile.linux b/doomgeneric/Makefile.linux
new file mode 100755
index 0000000..503e0dc
--- /dev/null
+++ b/doomgeneric/Makefile.linux
@@ -0,0 +1,54 @@
+################################################################
+#
+# $Id:$
+#
+# $Log:$
+#
+
+ifeq ($(V),1)
+	VB=''
+else
+	VB=@
+endif
+
+
+CC=clang  # gcc or g++
+CFLAGS+=-ggdb3 -Os
+LDFLAGS+=-Wl,--gc-sections
+CFLAGS+=-ggdb3 -Wall -DNORMALUNIX -DLINUX -DSNDSERV -D_DEFAULT_SOURCE # -DUSEASM
+LIBS+=-lm -lc -lX11
+
+# subdirectory for objects
+OBJDIR=build
+OUTPUT=doomgeneric
+
+SRC_DOOM = dummy.o am_map.o doomdef.o doomstat.o dstrings.o d_event.o d_items.o d_iwad.o d_loop.o d_main.o d_mode.o d_net.o f_finale.o f_wipe.o g_game.o hu_lib.o hu_stuff.o info.o i_cdmus.o i_endoom.o i_joystick.o i_scale.o i_sound.o i_system.o i_timer.o memio.o m_argv.o m_bbox.o m_cheat.o m_config.o m_controls.o m_fixed.o m_menu.o m_misc.o m_random.o p_ceilng.o p_doors.o p_enemy.o p_floor.o p_inter.o p_lights.o p_map.o p_maputl.o p_mobj.o p_plats.o p_pspr.o p_saveg.o p_setup.o p_sight.o p_spec.o p_switch.o p_telept.o p_tick.o p_user.o r_bsp.o r_data.o r_draw.o r_main.o r_plane.o r_segs.o r_sky.o r_things.o sha1.o sounds.o statdump.o st_lib.o st_stuff.o s_sound.o tables.o v_video.o wi_stuff.o w_checksum.o w_file.o w_main.o w_wad.o z_zone.o w_file_stdc.o i_input.o i_video.o doomgeneric.o doomgeneric_xlib.o
+OBJS += $(addprefix $(OBJDIR)/, $(SRC_DOOM))
+
+all:	 $(OUTPUT)
+
+clean:
+	rm -rf $(OBJDIR)
+	rm -f $(OUTPUT)
+	rm -f $(OUTPUT).gdb
+	rm -f $(OUTPUT).map
+
+$(OUTPUT):	$(OBJS)
+	@echo [Linking $@]
+	$(VB)$(CC) $(CFLAGS) $(LDFLAGS) $(OBJS) \
+	-o $(OUTPUT) $(LIBS) -Wl,-Map,$(OUTPUT).map
+	@echo [Size]
+	-$(CROSS_COMPILE)size $(OUTPUT)
+
+$(OBJS): | $(OBJDIR)
+
+$(OBJDIR):
+	mkdir -p $(OBJDIR)
+
+$(OBJDIR)/%.o:	%.c
+	@echo [Compiling $<]
+	$(VB)$(CC) $(CFLAGS) -c $< -o $@
+
+print:
+	@echo OBJS: $(OBJS)
+
diff --git a/doomgeneric/Makefile.sdl b/doomgeneric/Makefile.sdl
old mode 100644
new mode 100755
diff --git a/doomgeneric/Makefile.soso b/doomgeneric/Makefile.soso
old mode 100644
new mode 100755
diff --git a/doomgeneric/Makefile.sosox b/doomgeneric/Makefile.sosox
old mode 100644
new mode 100755
diff --git a/doomgeneric/am_map.c b/doomgeneric/am_map.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/am_map.h b/doomgeneric/am_map.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/config.h b/doomgeneric/config.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/d_englsh.h b/doomgeneric/d_englsh.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/d_event.c b/doomgeneric/d_event.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/d_event.h b/doomgeneric/d_event.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/d_items.c b/doomgeneric/d_items.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/d_items.h b/doomgeneric/d_items.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/d_iwad.c b/doomgeneric/d_iwad.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/d_iwad.h b/doomgeneric/d_iwad.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/d_loop.c b/doomgeneric/d_loop.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/d_loop.h b/doomgeneric/d_loop.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/d_main.c b/doomgeneric/d_main.c
old mode 100644
new mode 100755
index 9012e5f..682a555
--- a/doomgeneric/d_main.c
+++ b/doomgeneric/d_main.c
@@ -24,6 +24,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <unistd.h>
 
 #include "config.h"
 #include "deh_main.h"
diff --git a/doomgeneric/d_main.h b/doomgeneric/d_main.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/d_mode.c b/doomgeneric/d_mode.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/d_mode.h b/doomgeneric/d_mode.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/d_net.c b/doomgeneric/d_net.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/d_player.h b/doomgeneric/d_player.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/d_textur.h b/doomgeneric/d_textur.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/d_think.h b/doomgeneric/d_think.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/d_ticcmd.h b/doomgeneric/d_ticcmd.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/deh_main.h b/doomgeneric/deh_main.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/deh_misc.h b/doomgeneric/deh_misc.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/deh_str.h b/doomgeneric/deh_str.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/doom.h b/doomgeneric/doom.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/doomdata.h b/doomgeneric/doomdata.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/doomdef.c b/doomgeneric/doomdef.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/doomdef.h b/doomgeneric/doomdef.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/doomfeatures.h b/doomgeneric/doomfeatures.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/doomgeneric.c b/doomgeneric/doomgeneric.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/doomgeneric.h b/doomgeneric/doomgeneric.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/doomgeneric.vcxproj b/doomgeneric/doomgeneric.vcxproj
old mode 100644
new mode 100755
diff --git a/doomgeneric/doomgeneric.vcxproj.filters b/doomgeneric/doomgeneric.vcxproj.filters
old mode 100644
new mode 100755
diff --git a/doomgeneric/doomgeneric_allegro.c b/doomgeneric/doomgeneric_allegro.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/doomgeneric_emscripten.c b/doomgeneric/doomgeneric_emscripten.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/doomgeneric_ethereal.c b/doomgeneric/doomgeneric_ethereal.c
new file mode 100755
index 0000000..b125943
--- /dev/null
+++ b/doomgeneric/doomgeneric_ethereal.c
@@ -0,0 +1,158 @@
+/**
+ * @file doomgeneric/doomgeneric-_ethereal.c
+ * @brief Ethereal port
+ * 
+ * 
+ * @copyright
+ * This file is part of the Hexahedron kernel, which is part of reduceOS.
+ * It is released under the terms of the BSD 3-clause license.
+ * Please see the LICENSE file in the main repository for more details.
+ * 
+ * Copyright (C) 2024 Samuel Stuart
+ */
+
+#include "doomkeys.h"
+#include "m_argv.h"
+#include "doomgeneric.h"
+#include <stdio.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <kernel/gfx/video.h>
+#include <kernel/fs/periphfs.h>
+#include <sys/ioctl.h>
+#include <string.h>
+#include <ctype.h>
+#include <time.h>
+#include <ethereal/celestial.h>
+#include <ethereal/keyboard.h>
+
+keyboard_t *kbd = NULL;
+
+/* Allocated buffer for drawing */
+uint8_t *drawbuf = NULL;
+window_t *win = NULL;
+gfx_context_t *ctx = NULL;
+
+/* Key queue */
+const size_t key_queue_size = 16;
+unsigned long key_queue[16];
+size_t key_read_index = 0;
+size_t key_write_index = 0;
+
+unsigned char convertToDoomKey(int key);
+
+/* DOOM key event */
+void kbd_event(window_t *win, uint32_t event_type, void *event) {
+	celestial_event_key_t *key = (celestial_event_key_t*)event;
+	keyboard_event_t *ev = keyboard_event(kbd, &key->ev);
+	if (ev->scancode) {
+		key_queue[key_write_index] = convertToDoomKey(ev->scancode) | ((ev->type == KEYBOARD_EVENT_PRESS) << 8);
+		key_write_index = (key_write_index+1) % key_queue_size;
+	}
+}
+	
+
+void DG_Init() {
+    wid_t id = celestial_createWindow(0, DOOMGENERIC_RESX, DOOMGENERIC_RESY);
+    win = celestial_getWindow(id);
+    ctx = celestial_getGraphicsContext(win); 
+    celestial_setHandler(win, CELESTIAL_EVENT_KEY_EVENT, kbd_event);
+
+    gfx_clear(ctx, GFX_RGB(255, 255, 255));
+    gfx_render(ctx);
+    celestial_flip(win);
+
+    drawbuf = (uint8_t*)ctx->backbuffer;
+}
+
+void DG_DrawFrame() {
+    int i = 0;
+    uint8_t *buffer = drawbuf;
+    for (uint32_t y = 0; y < DOOMGENERIC_RESY; y++) {
+        for (uint32_t x = 0; x < DOOMGENERIC_RESX; x ++) {
+            ((uint32_t*)buffer)[x] = 0xFF000000 | DG_ScreenBuffer[i];
+            i++;
+        }
+
+        buffer += GFX_PITCH(ctx);
+    }
+
+    gfx_render(ctx);
+    celestial_flip(win);
+    celestial_poll();
+}
+
+
+void DG_SleepMs(uint32_t ms)
+{
+    usleep(ms * 1000);
+}
+
+uint32_t DG_GetTicksMs()
+{
+    struct timeval tp;
+    struct timezone tzp;
+    gettimeofday(&tp, &tzp);
+    return (tp.tv_sec * 1000) + (tp.tv_usec / 1000);
+}
+
+unsigned char convertToDoomKey(int key) {
+    switch (key) {
+        case '\t':
+            return KEY_TAB;
+        case '\n':
+            return KEY_ENTER;
+        case 'w':
+        case SCANCODE_UP_ARROW:
+            return KEY_UPARROW;
+        case 'a':
+        case SCANCODE_LEFT_ARROW:
+            return KEY_LEFTARROW;
+        case 's':
+        case SCANCODE_DOWN_ARROW:
+            return KEY_DOWNARROW;
+        case 'd':
+        case SCANCODE_RIGHT_ARROW:
+            return KEY_RIGHTARROW;
+        case SCANCODE_LEFT_CTRL:
+            return KEY_FIRE;
+        case ' ':
+            return KEY_USE;
+        case SCANCODE_ESCAPE:
+            return KEY_ESCAPE;
+        case SCANCODE_LEFT_ALT:
+            return KEY_LALT;
+        case SCANCODE_RIGHT_SHIFT:
+            return KEY_RSHIFT;
+        default:
+            return tolower(key);
+    }
+}
+
+
+
+
+int DG_GetKey(int* pressed, unsigned char* doomKey) {
+	if (key_read_index == key_write_index) return 0;
+
+	unsigned short key_data = key_queue[key_read_index];
+	key_read_index = (key_read_index + 1) % key_queue_size;
+	
+
+	*pressed = key_data >> 8;
+	*doomKey = key_data & 0xFF;
+	return 1;
+}
+
+void DG_SetWindowTitle(const char * title) {
+    celestial_setTitle(win, (char*)title);
+}
+
+
+
+int main(int argc, char *argv[]) {
+    kbd = keyboard_create();
+    doomgeneric_Create(argc, argv);
+    while (1) doomgeneric_Tick();
+    return 0;
+}
diff --git a/doomgeneric/doomgeneric_sdl.c b/doomgeneric/doomgeneric_sdl.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/doomgeneric_soso.c b/doomgeneric/doomgeneric_soso.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/doomgeneric_sosox.c b/doomgeneric/doomgeneric_sosox.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/doomgeneric_win.c b/doomgeneric/doomgeneric_win.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/doomgeneric_xlib.c b/doomgeneric/doomgeneric_xlib.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/doomkeys.h b/doomgeneric/doomkeys.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/doomstat.c b/doomgeneric/doomstat.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/doomstat.h b/doomgeneric/doomstat.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/doomtype.h b/doomgeneric/doomtype.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/dstrings.c b/doomgeneric/dstrings.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/dstrings.h b/doomgeneric/dstrings.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/dummy.c b/doomgeneric/dummy.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/f_finale.c b/doomgeneric/f_finale.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/f_finale.h b/doomgeneric/f_finale.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/f_wipe.c b/doomgeneric/f_wipe.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/f_wipe.h b/doomgeneric/f_wipe.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/g_game.c b/doomgeneric/g_game.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/g_game.h b/doomgeneric/g_game.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/gusconf.c b/doomgeneric/gusconf.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/gusconf.h b/doomgeneric/gusconf.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/hu_lib.c b/doomgeneric/hu_lib.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/hu_lib.h b/doomgeneric/hu_lib.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/hu_stuff.c b/doomgeneric/hu_stuff.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/hu_stuff.h b/doomgeneric/hu_stuff.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/i_allegromusic.c b/doomgeneric/i_allegromusic.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/i_allegrosound.c b/doomgeneric/i_allegrosound.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/i_cdmus.c b/doomgeneric/i_cdmus.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/i_cdmus.h b/doomgeneric/i_cdmus.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/i_endoom.c b/doomgeneric/i_endoom.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/i_endoom.h b/doomgeneric/i_endoom.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/i_input.c b/doomgeneric/i_input.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/i_joystick.c b/doomgeneric/i_joystick.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/i_joystick.h b/doomgeneric/i_joystick.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/i_scale.c b/doomgeneric/i_scale.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/i_scale.h b/doomgeneric/i_scale.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/i_sdlmusic.c b/doomgeneric/i_sdlmusic.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/i_sdlsound.c b/doomgeneric/i_sdlsound.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/i_sound.c b/doomgeneric/i_sound.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/i_sound.h b/doomgeneric/i_sound.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/i_swap.h b/doomgeneric/i_swap.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/i_system.c b/doomgeneric/i_system.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/i_system.h b/doomgeneric/i_system.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/i_timer.c b/doomgeneric/i_timer.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/i_timer.h b/doomgeneric/i_timer.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/i_video.c b/doomgeneric/i_video.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/i_video.h b/doomgeneric/i_video.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/icon.c b/doomgeneric/icon.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/info.c b/doomgeneric/info.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/info.h b/doomgeneric/info.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/m_argv.c b/doomgeneric/m_argv.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/m_argv.h b/doomgeneric/m_argv.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/m_bbox.c b/doomgeneric/m_bbox.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/m_bbox.h b/doomgeneric/m_bbox.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/m_cheat.c b/doomgeneric/m_cheat.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/m_cheat.h b/doomgeneric/m_cheat.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/m_config.c b/doomgeneric/m_config.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/m_config.h b/doomgeneric/m_config.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/m_controls.c b/doomgeneric/m_controls.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/m_controls.h b/doomgeneric/m_controls.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/m_fixed.c b/doomgeneric/m_fixed.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/m_fixed.h b/doomgeneric/m_fixed.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/m_menu.c b/doomgeneric/m_menu.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/m_menu.h b/doomgeneric/m_menu.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/m_misc.c b/doomgeneric/m_misc.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/m_misc.h b/doomgeneric/m_misc.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/m_random.c b/doomgeneric/m_random.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/m_random.h b/doomgeneric/m_random.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/memio.c b/doomgeneric/memio.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/memio.h b/doomgeneric/memio.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/mus2mid.c b/doomgeneric/mus2mid.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/mus2mid.h b/doomgeneric/mus2mid.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/net_client.h b/doomgeneric/net_client.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/net_dedicated.h b/doomgeneric/net_dedicated.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/net_defs.h b/doomgeneric/net_defs.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/net_gui.h b/doomgeneric/net_gui.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/net_io.h b/doomgeneric/net_io.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/net_loop.h b/doomgeneric/net_loop.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/net_packet.h b/doomgeneric/net_packet.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/net_query.h b/doomgeneric/net_query.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/net_sdl.h b/doomgeneric/net_sdl.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/net_server.h b/doomgeneric/net_server.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/p_ceilng.c b/doomgeneric/p_ceilng.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/p_doors.c b/doomgeneric/p_doors.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/p_enemy.c b/doomgeneric/p_enemy.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/p_floor.c b/doomgeneric/p_floor.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/p_inter.c b/doomgeneric/p_inter.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/p_inter.h b/doomgeneric/p_inter.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/p_lights.c b/doomgeneric/p_lights.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/p_local.h b/doomgeneric/p_local.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/p_map.c b/doomgeneric/p_map.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/p_maputl.c b/doomgeneric/p_maputl.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/p_mobj.c b/doomgeneric/p_mobj.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/p_mobj.h b/doomgeneric/p_mobj.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/p_plats.c b/doomgeneric/p_plats.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/p_pspr.c b/doomgeneric/p_pspr.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/p_pspr.h b/doomgeneric/p_pspr.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/p_saveg.c b/doomgeneric/p_saveg.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/p_saveg.h b/doomgeneric/p_saveg.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/p_setup.c b/doomgeneric/p_setup.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/p_setup.h b/doomgeneric/p_setup.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/p_sight.c b/doomgeneric/p_sight.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/p_spec.c b/doomgeneric/p_spec.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/p_spec.h b/doomgeneric/p_spec.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/p_switch.c b/doomgeneric/p_switch.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/p_telept.c b/doomgeneric/p_telept.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/p_tick.c b/doomgeneric/p_tick.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/p_tick.h b/doomgeneric/p_tick.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/p_user.c b/doomgeneric/p_user.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/r_bsp.c b/doomgeneric/r_bsp.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/r_bsp.h b/doomgeneric/r_bsp.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/r_data.c b/doomgeneric/r_data.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/r_data.h b/doomgeneric/r_data.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/r_defs.h b/doomgeneric/r_defs.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/r_draw.c b/doomgeneric/r_draw.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/r_draw.h b/doomgeneric/r_draw.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/r_local.h b/doomgeneric/r_local.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/r_main.c b/doomgeneric/r_main.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/r_main.h b/doomgeneric/r_main.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/r_plane.c b/doomgeneric/r_plane.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/r_plane.h b/doomgeneric/r_plane.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/r_segs.c b/doomgeneric/r_segs.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/r_segs.h b/doomgeneric/r_segs.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/r_sky.c b/doomgeneric/r_sky.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/r_sky.h b/doomgeneric/r_sky.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/r_state.h b/doomgeneric/r_state.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/r_things.c b/doomgeneric/r_things.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/r_things.h b/doomgeneric/r_things.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/s_sound.c b/doomgeneric/s_sound.c
old mode 100644
new mode 100755
index f6d8be1..399959c
--- a/doomgeneric/s_sound.c
+++ b/doomgeneric/s_sound.c
@@ -600,6 +600,9 @@ void S_StartMusic(int m_id)
 
 void S_ChangeMusic(int musicnum, int looping)
 {
+    // reduceOS best os
+    return;
+
     musicinfo_t *music = NULL;
     char namebuf[9];
     void *handle;
diff --git a/doomgeneric/s_sound.h b/doomgeneric/s_sound.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/sha1.c b/doomgeneric/sha1.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/sha1.h b/doomgeneric/sha1.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/sounds.c b/doomgeneric/sounds.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/sounds.h b/doomgeneric/sounds.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/st_lib.c b/doomgeneric/st_lib.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/st_lib.h b/doomgeneric/st_lib.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/st_stuff.c b/doomgeneric/st_stuff.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/st_stuff.h b/doomgeneric/st_stuff.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/statdump.c b/doomgeneric/statdump.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/statdump.h b/doomgeneric/statdump.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/tables.c b/doomgeneric/tables.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/tables.h b/doomgeneric/tables.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/v_patch.h b/doomgeneric/v_patch.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/v_video.c b/doomgeneric/v_video.c
old mode 100644
new mode 100755
index 6db28aa..510fd4b
--- a/doomgeneric/v_video.c
+++ b/doomgeneric/v_video.c
@@ -865,7 +865,7 @@ void V_DrawMouseSpeedBox(int speed)
     // If the mouse is turned off or acceleration is turned off, don't
     // draw the box at all.
 
-    if (!usemouse || fabs(mouse_acceleration - 1) < 0.01)
+    if (!usemouse || __builtin_fabs(mouse_acceleration - 1) < 0.01)
     {
         return;
     }
diff --git a/doomgeneric/v_video.h b/doomgeneric/v_video.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/w_checksum.c b/doomgeneric/w_checksum.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/w_checksum.h b/doomgeneric/w_checksum.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/w_file.c b/doomgeneric/w_file.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/w_file.h b/doomgeneric/w_file.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/w_file_stdc.c b/doomgeneric/w_file_stdc.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/w_main.c b/doomgeneric/w_main.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/w_main.h b/doomgeneric/w_main.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/w_merge.h b/doomgeneric/w_merge.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/w_wad.c b/doomgeneric/w_wad.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/w_wad.h b/doomgeneric/w_wad.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/wi_stuff.c b/doomgeneric/wi_stuff.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/wi_stuff.h b/doomgeneric/wi_stuff.h
old mode 100644
new mode 100755
diff --git a/doomgeneric/z_zone.c b/doomgeneric/z_zone.c
old mode 100644
new mode 100755
diff --git a/doomgeneric/z_zone.h b/doomgeneric/z_zone.h
old mode 100644
new mode 100755
diff --git a/screenshots/freebsd.png b/screenshots/freebsd.png
old mode 100644
new mode 100755
diff --git a/screenshots/sdl.png b/screenshots/sdl.png
old mode 100644
new mode 100755
diff --git a/screenshots/ubuntu.png b/screenshots/ubuntu.png
old mode 100644
new mode 100755
diff --git a/screenshots/windows.png b/screenshots/windows.png
old mode 100644
new mode 100755
