diff --git a/doomgeneric/Makefile b/doomgeneric/Makefile
index 503e0dc..2468966 100644
--- a/doomgeneric/Makefile
+++ b/doomgeneric/Makefile
@@ -12,17 +12,17 @@ else
 endif
 
 
-CC=clang  # gcc or g++
+CC=gcc
 CFLAGS+=-ggdb3 -Os
-LDFLAGS+=-Wl,--gc-sections
-CFLAGS+=-ggdb3 -Wall -DNORMALUNIX -DLINUX -DSNDSERV -D_DEFAULT_SOURCE # -DUSEASM
-LIBS+=-lm -lc -lX11
+# LDFLAGS+=-Wl,--gc-sections
+CFLAGS+=-ggdb3 -Wall $(shell pkg-config --cflags freetype2)  # -DUSEASM
+LIBS+= $(shell pkg-config --libs keyboard celestial) -lm -lc
 
 # subdirectory for objects
 OBJDIR=build
 OUTPUT=doomgeneric
 
-SRC_DOOM = dummy.o am_map.o doomdef.o doomstat.o dstrings.o d_event.o d_items.o d_iwad.o d_loop.o d_main.o d_mode.o d_net.o f_finale.o f_wipe.o g_game.o hu_lib.o hu_stuff.o info.o i_cdmus.o i_endoom.o i_joystick.o i_scale.o i_sound.o i_system.o i_timer.o memio.o m_argv.o m_bbox.o m_cheat.o m_config.o m_controls.o m_fixed.o m_menu.o m_misc.o m_random.o p_ceilng.o p_doors.o p_enemy.o p_floor.o p_inter.o p_lights.o p_map.o p_maputl.o p_mobj.o p_plats.o p_pspr.o p_saveg.o p_setup.o p_sight.o p_spec.o p_switch.o p_telept.o p_tick.o p_user.o r_bsp.o r_data.o r_draw.o r_main.o r_plane.o r_segs.o r_sky.o r_things.o sha1.o sounds.o statdump.o st_lib.o st_stuff.o s_sound.o tables.o v_video.o wi_stuff.o w_checksum.o w_file.o w_main.o w_wad.o z_zone.o w_file_stdc.o i_input.o i_video.o doomgeneric.o doomgeneric_xlib.o
+SRC_DOOM = dummy.o am_map.o doomdef.o doomstat.o dstrings.o d_event.o d_items.o d_iwad.o d_loop.o d_main.o d_mode.o d_net.o f_finale.o f_wipe.o g_game.o hu_lib.o hu_stuff.o info.o i_cdmus.o i_endoom.o i_joystick.o i_scale.o i_sound.o i_system.o i_timer.o memio.o m_argv.o m_bbox.o m_cheat.o m_config.o m_controls.o m_fixed.o m_menu.o m_misc.o m_random.o p_ceilng.o p_doors.o p_enemy.o p_floor.o p_inter.o p_lights.o p_map.o p_maputl.o p_mobj.o p_plats.o p_pspr.o p_saveg.o p_setup.o p_sight.o p_spec.o p_switch.o p_telept.o p_tick.o p_user.o r_bsp.o r_data.o r_draw.o r_main.o r_plane.o r_segs.o r_sky.o r_things.o sha1.o sounds.o statdump.o st_lib.o st_stuff.o s_sound.o tables.o v_video.o wi_stuff.o w_checksum.o w_file.o w_main.o w_wad.o z_zone.o w_file_stdc.o i_input.o i_video.o doomgeneric.o doomgeneric_ethereal.o
 OBJS += $(addprefix $(OBJDIR)/, $(SRC_DOOM))
 
 all:	 $(OUTPUT)
diff --git a/doomgeneric/doomgeneric_ethereal.c b/doomgeneric/doomgeneric_ethereal.c
new file mode 100755
index 0000000..cdc561e
--- /dev/null
+++ b/doomgeneric/doomgeneric_ethereal.c
@@ -0,0 +1,159 @@
+/**
+ * @file doomgeneric/doomgeneric-_ethereal.c
+ * @brief Ethereal port
+ * 
+ * 
+ * @copyright
+ * This file is part of the Hexahedron kernel, which is part of reduceOS.
+ * It is released under the terms of the BSD 3-clause license.
+ * Please see the LICENSE file in the main repository for more details.
+ * 
+ * Copyright (C) 2024 Samuel Stuart
+ */
+
+#include "doomkeys.h"
+#include "m_argv.h"
+#include "doomgeneric.h"
+#include <stdio.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <kernel/gfx/video.h>
+#include <kernel/fs/periphfs.h>
+#include <sys/ioctl.h>
+#include <string.h>
+#include <ctype.h>
+#include <time.h>
+#include <ethereal/celestial.h>
+#include <ethereal/keyboard.h>
+#include <sys/time.h>
+
+keyboard_t *kbd = NULL;
+
+/* Allocated buffer for drawing */
+uint8_t *drawbuf = NULL;
+window_t *win = NULL;
+gfx_context_t *ctx = NULL;
+
+/* Key queue */
+const size_t key_queue_size = 16;
+unsigned long key_queue[16];
+size_t key_read_index = 0;
+size_t key_write_index = 0;
+
+unsigned char convertToDoomKey(int key);
+
+/* DOOM key event */
+void kbd_event(window_t *win, uint32_t event_type, void *event) {
+	celestial_event_key_t *key = (celestial_event_key_t*)event;
+	keyboard_event_t *ev = keyboard_event(kbd, &key->ev);
+	if (ev->scancode) {
+		key_queue[key_write_index] = convertToDoomKey(ev->scancode) | ((ev->type == KEYBOARD_EVENT_PRESS) << 8);
+		key_write_index = (key_write_index+1) % key_queue_size;
+	}
+}
+	
+
+void DG_Init() {
+    wid_t id = celestial_createWindow(0, DOOMGENERIC_RESX, DOOMGENERIC_RESY);
+    win = celestial_getWindow(id);
+    ctx = celestial_getGraphicsContext(win); 
+    celestial_setHandler(win, CELESTIAL_EVENT_KEY_EVENT, kbd_event);
+
+    gfx_clear(ctx, GFX_RGB(255, 255, 255));
+    gfx_render(ctx);
+    celestial_flip(win);
+
+    drawbuf = (uint8_t*)ctx->backbuffer;
+}
+
+void DG_DrawFrame() {
+    int i = 0;
+    uint8_t *buffer = drawbuf;
+    for (uint32_t y = 0; y < DOOMGENERIC_RESY; y++) {
+        for (uint32_t x = 0; x < DOOMGENERIC_RESX; x ++) {
+            ((uint32_t*)buffer)[x] = 0xFF000000 | DG_ScreenBuffer[i];
+            i++;
+        }
+
+        buffer += GFX_PITCH(ctx);
+    }
+
+    gfx_render(ctx);
+    celestial_flip(win);
+    celestial_poll();
+}
+
+
+void DG_SleepMs(uint32_t ms)
+{
+    // usleep(ms);
+}
+
+uint32_t DG_GetTicksMs()
+{
+    struct timeval tp;
+    struct timezone tzp;
+    gettimeofday(&tp, &tzp);
+    return (tp.tv_sec * 1000) + (tp.tv_usec / 1000);
+}
+
+unsigned char convertToDoomKey(int key) {
+    switch (key) {
+        case '\t':
+            return KEY_TAB;
+        case '\n':
+            return KEY_ENTER;
+        case 'w':
+        case SCANCODE_UP_ARROW:
+            return KEY_UPARROW;
+        case 'a':
+        case SCANCODE_LEFT_ARROW:
+            return KEY_LEFTARROW;
+        case 's':
+        case SCANCODE_DOWN_ARROW:
+            return KEY_DOWNARROW;
+        case 'd':
+        case SCANCODE_RIGHT_ARROW:
+            return KEY_RIGHTARROW;
+        case SCANCODE_LEFT_CTRL:
+            return KEY_FIRE;
+        case ' ':
+            return KEY_USE;
+        case SCANCODE_ESCAPE:
+            return KEY_ESCAPE;
+        case SCANCODE_LEFT_ALT:
+            return KEY_LALT;
+        case SCANCODE_RIGHT_SHIFT:
+            return KEY_RSHIFT;
+        default:
+            return tolower(key);
+    }
+}
+
+
+
+
+int DG_GetKey(int* pressed, unsigned char* doomKey) {
+	if (key_read_index == key_write_index) return 0;
+
+	unsigned short key_data = key_queue[key_read_index];
+	key_read_index = (key_read_index + 1) % key_queue_size;
+	
+
+	*pressed = key_data >> 8;
+	*doomKey = key_data & 0xFF;
+	return 1;
+}
+
+void DG_SetWindowTitle(const char * title) {
+    celestial_setTitle(win, (char*)title);
+}
+
+
+
+int main(int argc, char *argv[]) {
+    kbd = keyboard_create();
+    doomgeneric_Create(argc, argv);
+    while (1) doomgeneric_Tick();
+    return 0;
+}
diff --git a/doomgeneric/s_sound.c b/doomgeneric/s_sound.c
index f6d8be1..c49acee 100644
--- a/doomgeneric/s_sound.c
+++ b/doomgeneric/s_sound.c
@@ -600,6 +600,9 @@ void S_StartMusic(int m_id)
 
 void S_ChangeMusic(int musicnum, int looping)
 {
+    // broken
+    return;
+
     musicinfo_t *music = NULL;
     char namebuf[9];
     void *handle;
