diff -Naur old_extract/libXrender-0.9.12/config.sub new_workspace/libXrender-0.9.12/config.sub
--- a/config.sub	2024-12-13 15:07:58.000000000 -0600
+++ b/config.sub	2026-06-23 16:02:00.537985122 -0500
@@ -1750,7 +1750,7 @@
 	     | clix* | riscos* | uniplus* | iris* | isc* | rtu* | xenix* \
 	     | mirbsd* | netbsd* | dicos* | openedition* | ose* \
 	     | bitrig* | openbsd* | secbsd* | solidbsd* | libertybsd* | os108* \
-	     | ekkobsd* | freebsd* | riscix* | lynxos* | os400* \
+	     | ekkobsd* | freebsd* | ethereal* | riscix* | lynxos* | os400* \
 	     | bosx* | nextstep* | cxux* | oabi* \
 	     | ptx* | ecoff* | winnt* | domain* | vsta* \
 	     | udi* | lites* | ieee* | go32* | aux* | hcos* \
diff -Naur old_extract/libXrender-0.9.12/configure new_workspace/libXrender-0.9.12/configure
--- a/configure	2024-12-13 15:07:57.000000000 -0600
+++ b/configure	2026-06-23 16:02:00.574214229 -0500
@@ -6095,6 +6095,10 @@
   lt_cv_deplibs_check_method=pass_all
   ;;
 
+ethereal*)
+      lt_cv_deplibs_check_method=pass_all
+      ;;
+
 hpux10.20* | hpux11*)
   lt_cv_file_magic_cmd=$FILECMD
   case $host_cpu in
@@ -11733,6 +11737,15 @@
   version_type=linux # correct to gnu/linux during the next big refactor
   need_lib_prefix=no
   need_version=no
+  library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
+  soname_spec='$libname$release$shared_ext$major'
+  shlibpath_var=LD_LIBRARY_PATH
+  ;;
+
+ethereal*)
+  version_type=linux # correct to gnu/linux during the next big refactor
+  need_lib_prefix=no
+  need_version=no
   library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
   soname_spec='$libname$release$shared_ext$major'
   shlibpath_var=LD_LIBRARY_PATH
