diff -Naur old_extract/libXScrnSaver-1.2.5/config.sub new_workspace/libXScrnSaver-1.2.5/config.sub
--- a/config.sub	2025-09-07 20:44:37.000000000 -0500
+++ b/config.sub	2026-06-23 16:01:59.846689134 -0500
@@ -2006,6 +2006,7 @@
 	| es* \
 	| fiwix* \
 	| freebsd* \
+	| ethereal* \
 	| fuchsia* \
 	| genix* \
 	| genode* \
diff -Naur old_extract/libXScrnSaver-1.2.5/configure new_workspace/libXScrnSaver-1.2.5/configure
--- a/configure	2025-09-07 20:44:36.000000000 -0500
+++ b/configure	2026-06-23 16:01:59.883211990 -0500
@@ -6238,6 +6238,10 @@
   lt_cv_deplibs_check_method=pass_all
   ;;
 
+ethereal*)
+      lt_cv_deplibs_check_method=pass_all
+      ;;
+
 hpux10.20* | hpux11*)
   lt_cv_file_magic_cmd=$FILECMD
   case $host_cpu in
@@ -11876,6 +11880,15 @@
   version_type=linux # correct to gnu/linux during the next big refactor
   need_lib_prefix=no
   need_version=no
+  library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
+  soname_spec='$libname$release$shared_ext$major'
+  shlibpath_var=LD_LIBRARY_PATH
+  ;;
+
+ethereal*)
+  version_type=linux # correct to gnu/linux during the next big refactor
+  need_lib_prefix=no
+  need_version=no
   library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
   soname_spec='$libname$release$shared_ext$major'
   shlibpath_var=LD_LIBRARY_PATH
