diff -Naur old_extract/libXmu-1.3.1/config.sub new_workspace/libXmu-1.3.1/config.sub
--- a/config.sub	2026-01-28 20:18:28.000000000 -0600
+++ b/config.sub	2026-06-23 16:02:00.057245920 -0500
@@ -2006,6 +2006,7 @@
 	| es* \
 	| fiwix* \
 	| freebsd* \
+	| ethereal* \
 	| fuchsia* \
 	| genix* \
 	| genode* \
diff -Naur old_extract/libXmu-1.3.1/configure new_workspace/libXmu-1.3.1/configure
--- a/configure	2026-01-28 20:18:26.000000000 -0600
+++ b/configure	2026-06-23 16:02:00.097212683 -0500
@@ -6533,6 +6533,10 @@
   lt_cv_deplibs_check_method=pass_all
   ;;
 
+ethereal*)
+      lt_cv_deplibs_check_method=pass_all
+      ;;
+
 hpux10.20* | hpux11*)
   lt_cv_file_magic_cmd=$FILECMD
   case $host_cpu in
@@ -12141,6 +12145,15 @@
   version_type=linux # correct to gnu/linux during the next big refactor
   need_lib_prefix=no
   need_version=no
+  library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
+  soname_spec='$libname$release$shared_ext$major'
+  shlibpath_var=LD_LIBRARY_PATH
+  ;;
+
+ethereal*)
+  version_type=linux # correct to gnu/linux during the next big refactor
+  need_lib_prefix=no
+  need_version=no
   library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
   soname_spec='$libname$release$shared_ext$major'
   shlibpath_var=LD_LIBRARY_PATH
