diff -Naur old_extract/libxcb-1.17.0/build-aux/config.sub new_workspace/libxcb-1.17.0/build-aux/config.sub
--- a/build-aux/config.sub	2024-04-15 10:09:35.000000000 -0500
+++ b/build-aux/config.sub	2026-06-23 15:10:40.524415339 -0500
@@ -1718,7 +1718,7 @@
 	     | clix* | riscos* | uniplus* | iris* | isc* | rtu* | xenix* \
 	     | mirbsd* | netbsd* | dicos* | openedition* | ose* \
 	     | bitrig* | openbsd* | secbsd* | solidbsd* | libertybsd* | os108* \
-	     | ekkobsd* | freebsd* | riscix* | lynxos* | os400* \
+	     | ekkobsd* | freebsd* | ethereal* | riscix* | lynxos* | os400* \
 	     | bosx* | nextstep* | cxux* | aout* | elf* | oabi* \
 	     | ptx* | coff* | ecoff* | winnt* | domain* | vsta* \
 	     | udi* | lites* | ieee* | go32* | aux* | hcos* \
diff -Naur old_extract/libxcb-1.17.0/configure new_workspace/libxcb-1.17.0/configure
--- a/configure	2024-04-15 10:09:34.000000000 -0500
+++ b/configure	2026-06-23 15:10:46.857649876 -0500
@@ -6946,6 +6946,10 @@
   lt_cv_deplibs_check_method=pass_all
   ;;
 
+ethereal*)
+      lt_cv_deplibs_check_method=pass_all
+      ;;
+
 hpux10.20* | hpux11*)
   lt_cv_file_magic_cmd=$FILECMD
   case $host_cpu in
@@ -12687,6 +12691,15 @@
   version_type=linux # correct to gnu/linux during the next big refactor
   need_lib_prefix=no
   need_version=no
+  library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
+  soname_spec='$libname$release$shared_ext$major'
+  shlibpath_var=LD_LIBRARY_PATH
+  ;;
+
+ethereal*)
+  version_type=linux # correct to gnu/linux during the next big refactor
+  need_lib_prefix=no
+  need_version=no
   library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
   soname_spec='$libname$release$shared_ext$major'
   shlibpath_var=LD_LIBRARY_PATH
